shoulder/generator/c_header_generator.py

```diff
@@ -365,9 +383,17 @@ def _generate_vmsr_set(self, outfile, reg, am):
 
     @shoulder.gadget.c.function_definition
     def _generate_str_set(self, outfile, reg, am):
-        reg_setter = "SHOULDER_AARCH64_STR_IMPL({addr}, val)".format(
-            addr="0x0"
+        reg_setter =  "if ({base} == 0x0) return 0xFFFFFFFFFFFFFFFF;\n".format(
+            base=str(am.component.upper()) + "_BASE"
         )
+
+        reg_setter += "{size} addr = {base} + {offset};\n".format(
```

Comment:
This makes me realize something dumb that I did in my last PR: I split the LDR/STR accessor macros into two different files. aarch32 ones that use uint32_t for addresses, and aarch64 ones that use uint64_t for addresses. We should probably just have one set that uses ``uintptr_t``` for addresses.

I think that ```addr``` here needs to be a ```uintptr_t``` as well, because you can have a 32-bit register that is located at a 64-bit address (and vice-versa)

Alternately, you could generate ```SHOULDER_AARCH64_STR_IMPL(<BASE_ADDR> + <offset>, val);```

Reply:
Ok, I can combine them into one macro that uses `uintptr_t`. Should I create a new macros file in the `include/` directory since these external accesses aren't necessarily 32bit or 64bit exclusive?

Also, I will alter the function definition to just do the base + offset in the macro argument passing step.